### DISCUSSION TITLE
How can I configure a door to open only during specific time frame?

### DISCUSSION THREAD

**Discussion by taufpate** [uhppoted/uhppoted#9] [Q&A] [answered]
Hi! i have múltiple questions:

1.- How can I achieve that one door opens only in a specific time frame? (for one or all cards)

2.- in "addTask" method the "cards" key its the card index? what about if it's multiple cards or all cards?

**Comment by uhppoted** [comment, marked as answer]
Hi,

1. Time profiles allow you to set the days and periods during which a door can be opened. You can set combinations of specific days e.g. Monday, Wednesday, Friday and up to 3 segments per profile e.g. 08:00 to 11:30, 14:30 to 17:45, etc. And you can link multiple time profiles if you need more combinations. You then update the card records to use the time profile for the specific door. Ok, that's a bit unclear - let me give an example rather... 

   Assuming you want to enable door 4 only between 08:30 and 17:00 on weekdays for staff, you would first create a time    profile:
```
const deviceID = 405419896
const profile = {
  id: 29,
  valid: { from: '2022-01-01', to: '2022-12-31' },
  weekdays: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday'],
  segments: [
    { start: '08:30', end: '17:00' }
  ]
}

uhppoted.setTimeProfile(ctx, deviceID, profile)
...
...
```

You would then assign the profile to door 4 for all the staff cards:

```
const validFrom = '2022-01-01'
const validUntil = '2022-12-31'
const doors = { 1: true, 2: false, 3: true, 4: 29 }

staff.forEach(card => {
   uhppoted.putCard(ctx, deviceID, card, validFrom, validUntil, doors)
   ...
   ...
}
```

2. The `cards` field in `AddTask` is for use with the `enable more cards` task and sets the number of people allowed access after the privileged _more card_ has swiped. AFAIK, it's primarily for things like escorting guests or guided tours - so e.g. if you were guiding a tour of 12 people your card would be assigned the _more cards_ privilege and `AddTask` + _enable more cards_ would be used to limit the number of people that could enter after you. I'm afraid that's about as much information as I have at the moment (it's not something I use). There are also a couple of additional functions on the TODO list around _more card_ and _first card_ privileges that would have to be implemented before it could really be used.

**Comment by taufpate** [comment]
once again thanks for your quick response and amazing work!, i'll try what you say and let you know

**Comment by uhppoted** [comment]
hiya - moved this to **Q&A** under _Discussions_ because it's come up once or twice before. Going to mark it as answered - feel free to add/ask if you have anything more.

**Comment by uhppoted** [comment]
From the [Amazon Q&A page](https://www.amazon.com/ask/questions/Tx38886TQ1ALP67/ref=ask_dp_dpmw_al_hza):

> The latest version from the wiegand.com.cn site is version 7.111 and it just came out January. The "Task-List" is what you want if you want the doors to change to open at 8:00 AM (for example) The "Time Profiles" is only for narrowing the time that users can open the doors with the readers. Note: There is no "Holiday Over-Ride in the software. You set your schedule up from now until the next holiday. Then you start the next schedule up after the the holiday.
> ...
> ...
> I use Professionnal Door Control Management v 7.75.87.51213 (i think it is the lastest version) and tried to get TIME PROFILES working.
> For sure there is a bug : i can't get an access granted from a swpe within a specified time frame.
> The documentation shows a time frame should be linked to Time Profile 0 but no way , … [see more](https://www.amazon.com/ask/questions/Tx38886TQ1ALP67/ref=ask_dp_dpmw_al_hza#)
> ...
> ...
> Yes - Add "Controller Task List" through the Extended login. Create a task to open and close at certain times! Phew!! Took me a long time to figure that one out...
> ...
> ...
> yes,you coukd control the door by time schedule.thank you.
> ...
> ...

**Comment by uhppoted** [comment]
Moved to the [top level discussions page](https://github.com/uhppoted/uhppoted/discussions), because it's a general interest item.
